Move `demangle` helper to library_legacy.js (#21346)
emscripten-core/emscripten: ChangeLog.md

3.1.54 (in development)
-----------------------

> - The `DEMANGLE_SUPPORT` setting and the associated `demangle` function are
>   now deprecated since Wasm stack traces no longer contain mangled symbol
>   names under any circumstances. (#21346)

Reviewer: ```suggestion
- The `DEMANGLE_SUPPORT` setting and the associated `demangle` function are
  now deprecated since Wasm stack traces are always demangled now anyhow. (#21346)
```
